ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -82,16 +86,105 @@ import           Control.Concurrent.STM.Stats         (atomically, modifyTVar',
                                                        readTVar, writeTVar)
 import           Control.Concurrent.STM.TQueue
 import           Data.Foldable                        (for_)
+import           Data.HashMap.Strict                  (HashMap)
+import           Data.HashSet                         (HashSet)
 import qualified Data.HashSet                         as Set
 import           Database.SQLite.Simple
 import           Development.IDE.Core.Tracing         (withTrace)
 import           Development.IDE.Types.Shake          (WithHieDb)
 import           HieDb.Create
 import           HieDb.Types
 import           HieDb.Utils
+import           Prettyprinter                        (Pretty (pretty), (<+>))
+import qualified Prettyprinter
 import           System.Random                        (RandomGen)
 import qualified System.Random                        as Random
 
+data Log
+  = LogSettingInitialDynFlags
+  | LogGetInitialGhcLibDirDefaultCradleFail !CradleError !FilePath !(Maybe FilePath) !(Cradle Void)
+  | LogGetInitialGhcLibDirDefaultCradleNone
+  | LogHieDbRetry !Int !Int !Int !SomeException
+  | LogHieDbRetriesExhausted !Int !Int !Int !SomeException
+  | LogHieDbWriterThreadSQLiteError !SQLError
+  | LogHieDbWriterThreadException !SomeException
+  | LogInterfaceFilesCacheDir !FilePath
+  | LogKnownFilesUpdated !(HashMap Target (HashSet NormalizedFilePath))
+  | LogMakingNewHscEnv ![UnitId]
+  | LogDLLLoadError !String
+  | LogCradlePath !FilePath
+  | LogCradleNotFound !FilePath
+  | LogSessionLoadingResult !(Either [CradleError] (ComponentOptions, FilePath))
+  | forall a. Show a => LogCradle !(Cradle a)
```

Comment:
I think it should be fine to choose here the concrete cradle

```suggestion
  | LogCradle !(Cradle Void)
```

Reply:
Done, but that also required changing `cradleToOptAndLibDir` to take a `Cradle Void` instead of a `Show a => Cradle a` as it did previously.